api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/EriDSDCohortQueries.java

```diff
@@ -413,7 +414,13 @@ public CohortDefinition getPatientsWhoAreActiveAndUnstable() {
             getAllPatientsWhoAreActiveAndStable(),
             "startDate=${startDate},endDate=${endDate},location=${location}"));
 
-    cd.setCompositionString("allPatientsTxCurr AND NOT activeAndStablePatients");
+    cd.addSearch(
+        "tbPatient",
+        EptsReportUtils.map(
+            txTBCohortQueries.getInTBProgram(),
+            "startDate=${startDate},endDate=${endDate},location=${location}"));
+
+    cd.setCompositionString("allPatientsTxCurr AND NOT (activeAndStablePatients OR tbPatient)");
```

Comment:
Its missing pregnant and breastfeeding exclusions.

Reply:
not sure!  Because  `activeAndStablePatients` already have pregnant and breastfeeding